### PR TITLE
Outputs ng

### PR DESCRIPTION
**Pull request by mikouaj**
Introduced evaluation outputs that are inspired by `gcpdiag`. 
* output mapper and related funcs are replaced with `outputs.ValidationReport`
* the structure is now policy oriented rather than input data (cluster data) oriented
* due to above, `policy.PolicyEvaluationResult` was simplified to contain simple slice of policies only *(no groupings etc)*
* the output still contains colors and some icons where added as well ;-)

### CODE REVIEW

File: internal/outputs/console_collector.go [hunk 1]
```diff
@@ -18,38 +18,49 @@ import (
...
-			result.ErroredCount())
+		p.out.TabFlush()
+		p.out.Printf("\n")
 	}
-
-	return nil
-}
-
-func (p *ConsoleResultCollector) Close() error {
+	p.out.ColorPrintf("\u2139 [white][bold]Evaluated %d policies on %d clusters\n", len(report.Policies), len(report.ClusterStats))
```
Comment (ewojtach) [inline internal/outputs/console_collector.go:59]: white texts are not visible on white terminals ;) maybe we can change to grey?

File: internal/app/app.go [hunk 2]
```diff
@@ -221,7 +221,7 @@ func (p *PolicyAutomationApp) evaluateClusters(regoPackageBases []string) error
 	}
 	// create a PolicyAgent client instance
 	pa := policy.NewPolicyAgent(p.ctx)
-	p.out.ColorPrintf("[light_gray][bold]Parsing REGO policies...\n")
+	p.out.ColorPrintf("\u2139 [light_gray][bold]Parsing REGO policies...\n")
```
Comment (ewojtach) [inline internal/app/app.go:224]: maybe we can define some named const values for icons?

Comment (mikouaj) [reply internal/app/app.go:224]: Good idea, I've added `icons.go` with consts starting with `ICON_xx`.
However in future we should think about automatic replacement method as we have now with colors :-)

File: internal/outputs/console_collector.go [hunk 3]
```diff
@@ -18,38 +18,49 @@ import (
...
+	p.reportMapper.AddResults(results)
+	return nil
+}
+
+func (p *consoleResultCollector) Close() error {
+	report := p.reportMapper.GetReport()
+	p.out.Printf("\n")
+	p.out.InitTabs(95)
+	for _, policy := range report.Policies {
+		p.out.ColorPrintf("\U0001f50e [bold][white][%s][yellow] %s[reset]: %s\n", policy.PolicyGroup, policy.PolicyName, policy.PolicyTitle)
```
[the remark this answers is not in the document]
Comment (mikouaj) [reply internal/outputs/console_collector.go:43]: I agree, light_grey on the way! :)

File: internal/outputs/validation_report.go [hunk 4]
```diff
@@ -0,0 +1,163 @@
...
+	return &validationReportMapperImpl{
+		policies:        make(map[string]*ValidationReportPolicy),
+		clusterStats:    make(map[string]*ValidationReportClusterStats),
+		jsonMarshalFunc: json.Marshal,
+		validationTime:  time.Now(),
+	}
+}
+
+func (m *validationReportMapperImpl) AddResult(result *policy.PolicyEvaluationResult) {
+	clusterStat, ok := m.clusterStats[result.ClusterName]
```
[the remark this answers is not in the document]
Comment (mikouaj) [reply internal/outputs/validation_report.go:78]: Good comment, indeed there is an `ID` there. I've changed accordingly.

**Pull request merged**